presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoResultSet.java

```diff
@@ -92,7 +92,20 @@
             .toFormatter()
             .withOffsetParsed();
 
-    static final DateTimeFormatter TIMESTAMP_FORMATTER = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss.SSS");
+    static final DateTimeFormatter TIMESTAMP_FORMATTER = new DateTimeFormatterBuilder()
+            .append(
+                    DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss.SSS").getPrinter(),
+                    new DateTimeParser[] {
+                            DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss").getParser(),
+                            DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss.S").getParser(),
+                            DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss.SS").getParser(),
+                            DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss.SSS").getParser(),
+                            DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss.SSSS").getParser(),
+                            DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss.SSSSS").getParser(),
+                            DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss.SSSSSS").getParser(),
```

Comment:
don't you need up to 12?

Reply:
Yes, but that actually doesn't help anyway. Joda only supports parsing up to millisecond precision, so everything after that is lost. I changed the parser entirely to support up to nanosecond precision (what JDBC Timestamp allows).